### PR TITLE
Allow create/update endpoint registry entry when no definition file/url provided

### PR DESCRIPTION
**Pull request by ChamodDamitha**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/util/EndpointRegistryMappingUtils.java [hunk 1]
```diff
@@ -43,8 +43,12 @@ public static EndpointRegistryInfo fromDTOtoEndpointRegistry(RegistryDTO registr
         EndpointRegistryInfo registry = new EndpointRegistryInfo();
         registry.setName(registryDTO.getName());
         registry.setOwner(owner);
-        registry.setType(registryDTO.getType().toString());
-        registry.setMode(registryDTO.getMode().toString());
+        if (registryDTO.getType() != null) {
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/util/EndpointRegistryMappingUtils.java:46]: set defaults if not provided

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java [hunk 2]
```diff
@@ -817,6 +856,57 @@ public void updateRegistryEntryWithDefinitionUrl() throws APIManagementException
         compareRegistryEntryDTOs(payloadEntryDTO, responseEntryDTO);
     }
 
+    @Test
+    public void updateRegistryEntryWithNoDefinitionFileAndUrl() throws APIManagementException {
+        final String REGISTRY_UUID = "reg1";
+
+        EndpointRegistryInfo endpointRegistryInfo = new EndpointRegistryInfo();
```
Comment (fazlan-nazeem) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java:863]: shall we introduce a method to create a registryEntry when parameters are given? also another method to return a registryEntry with some default values.

Comment (fazlan-nazeem) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java:863]: the reason is, if we are to add a new property to a schema there;s a lot of places to change to include them in the tests. By introducing a method we can minimize that effort.

Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java:863]: sure, will add that

Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java]: Added in https://github.com/wso2/carbon-apimgt/pull/8552/commits/e168f5df88c3d0fd01b64f4ad08d69c5bed0b45e

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/util/EndpointRegistryMappingUtils.java (hunk 1 above)
Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/util/EndpointRegistryMappingUtils.java:46]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java [hunk 3]
```diff
@@ -263,23 +228,12 @@ public void addRegistry_ResourceNameExists() throws APIManagementException {
...
-        payloadDTO.setType(RegistryDTO.TypeEnum.WSO2);
-        payloadDTO.setOwner(ADMIN_USERNAME);
-
-        EndpointRegistryInfo endpointRegistryInfoNew =
-                EndpointRegistryMappingUtils.fromDTOtoEndpointRegistry(payloadDTO, ADMIN_USERNAME);
-        endpointRegistryInfoNew.setUuid("abc1");
+        EndpointRegistryInfo endpointRegistryInfoOld = createRegistryWithDefaultParams();
+        EndpointRegistryInfo endpointRegistryInfoNew = createRegistry(endpointRegistryInfoOld.getUuid(), 2,
+                "Endpoint Registry 2", RegistryDTO.ModeEnum.READWRITE, RegistryDTO.TypeEnum.ETCD,
+                "user1");
```
Comment (Meruja) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java:234]: Shall we move "user1" to line 233?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java [hunk 4]
```diff
@@ -297,23 +251,12 @@ public void updateRegistry() throws APIManagementException {
...
+
```
Comment (Meruja) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java:258]: Shall we move "user1" to line 256?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java [hunk 5]
```diff
@@ -914,26 +685,18 @@ public void getAllEntriesInRegistry() throws APIManagementException {
...
-        endpointRegistryEntry2.setEntryId("entry2");
-        endpointRegistryEntry2.setName("Entry Name 2");
-        endpointRegistryEntry2.setMetaData("{mutualTLS: true}");
-        endpointRegistryEntry2.setProductionServiceURL("https://xyz2.com");
-        endpointRegistryEntry2.setServiceType(RegistryEntryDTO.ServiceTypeEnum.SOAP_1_1.toString());
-        endpointRegistryEntry2.setServiceCategory(RegistryEntryDTO.ServiceCategoryEnum.DOMAIN.toString());
-        endpointRegistryEntry2.setDefinitionURL("https://petstore.swagger.io/v2/swagger2.json");
-        endpointRegistryEntry2.setDefinitionType(RegistryEntryDTO.DefinitionTypeEnum.GQL_SDL.toString());
+        EndpointRegistryEntry endpointRegistryEntry2 = createRegistryEntry("entry2", "Entry Name 2",
+                "{mutualTLS: false}", "https://xyz2.com",
```
Comment (Meruja) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java:696]: Shall we declare variables for "Entry Name 2", "{mutualTLS: false}" and "https://xyz2.com"?
It has been used in many places.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java (hunk 3 above)
Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java:234]: Removed the file in https://github.com/wso2/carbon-apimgt/pull/8615

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java (hunk 4 above)
Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java:258]: Removed the file in https://github.com/wso2/carbon-apimgt/pull/8615

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java (hunk 5 above)
Comment (ChamodDamitha) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/test/java/org.wso2.carbon.apimgt.rest.api.endpoint.registry/impl/RegistriesApiServiceImplTest.java:696]: Removed the file in https://github.com/wso2/carbon-apimgt/pull/8615

**Pull request merged**